build_tools/configure_ci_test.py

```diff
@@ -29,6 +30,54 @@ def test_run_ci_if_source_file_and_unrelated_workflow_file_edited(self):
         run_ci = configure_ci.should_ci_run_given_modified_paths(paths)
         self.assertTrue(run_ci)
 
+    @mock.patch.dict(
+        os.environ,
+        {
+            "INPUT_LINUX_AMDGPU_FAMILIES": "   gfx94X ,|.\\,  gfx1201X, --   gfx90X",
+            "INPUT_WINDOWS_AMDGPU_FAMILIES": "gfx94X \\., gfx1201X  gfx90X",
+        },
+    )
```

Comment:
Nice, thanks for adding tests using the new inputs.

Another thing we could do for testing would be to have `main()` in `configure_ci.py` take a dictionary of variables and populate those variables in `if __name__ == "__main__":`, then have the test script call `main` with a manually constructed dictionary instead of using environment variables. A similar pattern is also used when having a script use argparse when run via CLI and explicit arguments when run as a library.

```py
def configure_ci(args):
  ...

if __name__ == "__main__":
    args = {}
    args.linux_amdgpu_families = os.environ.get("INPUT_LINUX_AMDGPU_FAMILIES", "")
    configure_ci(args)
```